### PR TITLE
soc: esp32c2: Added support to ESP32C2 and ESP8684

### PR DESCRIPTION
**Pull request by raffarost**
Added support to ESP32C2 and ESP8684 (same core).

Initial features:

- Simple boot support
- Basic drivers:
    - Clock control
    - UART
    - GPIO
    - Watchdog

### CODE REVIEW

**Review by gmarull — CHANGES_REQUESTED** [review]
please split combo commits like e56ba23ce2e09101ca1ae511675e5982713d311e properly to ease review.

**Review by uLipe — CHANGES_REQUESTED** [review]
In general LGTM, my only concern are the soc support and dts addition commits, I think they would be combined into a single one since no SoC support comes with dts or soc code support, so I suggest to combine:

91cc4a761a23bb6a0e2518272d7c9dc11d4ca14d

and

91cc4a761a23bb6a0e2518272d7c9dc11d4ca14d

into a single commit in this patchset.

**Review by gmarull — DISMISSED** [review]
please, do not create combo commits like c941736f52f71f07f19de472c2cf32d21db84c86

File: boards/common/esp32.board.cmake [hunk 1]
```diff
@@ -13,4 +13,4 @@ board_runner_args(openocd  --gdb-init "thb main")
 set(ESP_IDF_PATH ${ZEPHYR_HAL_ESPRESSIF_MODULE_DIR})
 assert(ESP_IDF_PATH "ESP_IDF_PATH is not set")
 
-board_finalize_runner_args(esp32 "--esp-idf-path=${ESP_IDF_PATH}")
+board_runner_args(esp32 "--esp-idf-path=${ESP_IDF_PATH}")
```
Comment (marekmatej) [inline boards/common/esp32.board.cmake:16]: Please do not mix this with new board support PRs.

**Review by marekmatej — CHANGES_REQUESTED** [review]
Split generic stuff from the particular board support.

**Comment by marekmatej** [comment]
I believe splitting the support for ESP32-C2 and ESP-8684 into separate PRs would help to speed up the review process! Regardless the fact they share the same features or not.

**Comment by raffarost** [comment]
> I believe splitting the support for ESP32-C2 and ESP-8684 into separate PRs would help to speed up the review process! Regardless the fact they share the same features or not.

The chip is the same, only difference being ESP8684 has in-chip flash. Functionally they are the same for all contents in the PR

File: scripts/west_commands/runners/esp32.py [hunk 2]
```diff
@@ -62,6 +62,8 @@ def do_add_parser(cls, parser):
                             help='serial port to flash')
         parser.add_argument('--esp-baud-rate', default='921600',
                             help='serial baud rate, default 921600')
+        parser.add_argument('--esp-monitor-baud', default='115200',
```
Comment (marekmatej) [inline scripts/west_commands/runners/esp32.py:65]: can you please move this change to a separate PR as it is not related to the C2 support?

**Review by marekmatej — CHANGES_REQUESTED** [review]

File: scripts/west_commands/runners/esp32.py (hunk 2 above)
Comment (raffarost) [reply scripts/west_commands/runners/esp32.py:65]: This is actually for C2. The standard baud rate is 74880. so we need to pass this parameter to `monitor`

**Review by LucasTambor — DISMISSED** [review]

**Review by marekmatej — DISMISSED** [review]

File: soc/espressif/esp32c2/Kconfig [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+# SPDX-License-Identifier: Apache-2.0
+
+config SOC_SERIES_ESP32C2
+	select RISCV
+	select RISCV_GP
+	select DYNAMIC_INTERRUPTS
+	select CLOCK_CONTROL
+	select PINCTRL
+	select RISCV_ISA_RV32I
+	select RISCV_ISA_EXT_A
```
Comment (sylvioalves) [inline soc/espressif/esp32c2/Kconfig:11]: I think C2 has no Atomic set. PLease confirm and remove `select RISCV_ISA_EXT_A`.

File: soc/espressif/esp32c2/Kconfig [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+	select RISCV_ISA_EXT_M
+	select RISCV_ISA_EXT_C
+	select RISCV_ISA_EXT_ZICSR
+	select HAS_ESPRESSIF_HAL
+	select XIP if !MCUBOOT
+
+if SOC_SERIES_ESP32C2
+
+config ESP32_PHY_MAX_WIFI_TX_POWER
```
Comment (sylvioalves) [inline soc/espressif/esp32c2/Kconfig:20]: This (and other) has been moved into `Kconfig.wifi`. Not sure if merged already.

File: soc/espressif/esp32c2/soc.h [hunk 5]
```diff
@@ -0,0 +1,56 @@
...
+#include <zephyr/types.h>
+#include <stdbool.h>
+#include <esp_private/esp_clk.h>
+#endif
+
+#ifndef _ASMLANGUAGE
+
+void __esp_platform_start(void);
+
+static inline uint32_t esp_core_id(void)
```
Comment (sylvioalves) [inline soc/espressif/esp32c2/soc.h:23]: Can we remove this? Either use Zephyr's arch API or cpu_ll.h if needed.

Comment (raffarost) [reply soc/espressif/esp32c2/soc.h:23]: This is used in a couple of drivers and is present in all other SoC's.
Should we do a separate PR?

**Review by sylvioalves — APPROVED** [review]

**Review by LucasTambor — APPROVED** [review]

**Review by uLipe — APPROVED** [review]

**Review by marekmatej — APPROVED** [review]

**Comment by marekmatej** [comment]
@nashif could you take a look?
